### PR TITLE
Convert RandomTranslation to the backend-agnostic implementation

### PR DESCRIPTION
**Pull request by james77777778**
Related to #481 

Tests all passed in my local env

### CODE REVIEW

File: keras_core/layers/preprocessing/random_translation.py [hunk 1]
```diff
@@ -73,43 +90,157 @@ def __init__(
...
+        super().__init__(**kwargs)
+        self.height_factor = height_factor
+        self.height_lower, self.height_upper = self._set_factor(
+            height_factor, "height_factor"
         )
+        self.width_factor = width_factor
+        self.width_lower, self.width_upper = self._set_factor(
+            width_factor, "width_factor"
+        )
+        self._check_fill_mode_and_interpolation(fill_mode, interpolation)
```
Comment (fchollet) [inline keras_core/layers/preprocessing/random_translation.py:105]: You can inline this logic here, no need for a separate method since it's only used 1x

**Review by fchollet — COMMENTED** [review]
Thanks for the PR. It's looking good! Please make sure we have enough test coverage, I think the current unit tests are pretty weak. You can use `pytest-cov` to check which parts of the code are covered.

File: keras_core/layers/preprocessing/random_translation.py (hunk 1 above)
Comment (james77777778) [reply keras_core/layers/preprocessing/random_translation.py]: Done

**Comment by james77777778** [comment]
I have added some tests including:
- raise ValueError with bad factors
- training=False behavior
- channels_first & channels_last

Now the coverage becomes 100%
```bash
Name                                                    Stmts   Miss  Cover
---------------------------------------------------------------------------
keras_core/layers/preprocessing/random_translation.py      74      0   100%
```

**Review by fchollet — APPROVED** [review]
Excellent -- thank you for the contribution!

**Pull request merged**